include/extractor/guidance/validation_handler.hpp

```diff
@@ -0,0 +1,86 @@
+#ifndef OSRM_EXTRACTOR_GUIDANCE_VALIDATION_HANDLER_HPP_
+#define OSRM_EXTRACTOR_GUIDANCE_VALIDATION_HANDLER_HPP_
+
+#include "extractor/guidance/intersection.hpp"
+#include "extractor/guidance/intersection_generator.hpp"
+#include "extractor/guidance/intersection_handler.hpp"
+#include "extractor/query_node.hpp"
+
+#include "util/name_table.hpp"
+#include "util/node_based_graph.hpp"
+
+#include <vector>
+
+namespace osrm
+{
+namespace extractor
+{
+namespace guidance
+{
+
+// Runs sanity checks on intersections and dumps out suspicious ones.
+class ValidationHandler final : public IntersectionHandler
+{
+  public:
+    ValidationHandler(const IntersectionGenerator &intersection_generator,
+                      const util::NodeBasedDynamicGraph &node_based_graph,
+                      const std::vector<util::Coordinate> &coordinates,
+                      const util::NameTable &name_table,
+                      const SuffixTable &street_name_suffix_table)
+        : IntersectionHandler(node_based_graph,
+                              coordinates,
+                              name_table,
+                              street_name_suffix_table,
+                              intersection_generator)
+    {
+    }
+
+    ~ValidationHandler() override final = default;
+
+    bool canProcess(const NodeID, const EdgeID, const Intersection &) const override final
+    {
+        return true;
+    }
+
+    Intersection operator()(const NodeID nid,
+                            const EdgeID via_eid,
+                            Intersection intersection) const override final
+    {
+        BOOST_ASSERT(intersection.size() > 2);
+
+        checkForSharpTurnsOntoRamps(nid, via_eid, intersection);
+
+        return intersection;
+    }
+
+  private:
+    void checkForSharpTurnsOntoRamps(const NodeID,
+                                     const EdgeID via_eid,
+                                     const Intersection &intersection) const
+    {
+        for (std::size_t i = 1; i < intersection.size(); ++i)
+        {
+            const auto road = intersection[i];
+
+            if (!(road.instruction.type == TurnType::OnRamp) || !road.entry_allowed)
```

Comment:
@daniel-j-h what is your opinion on interaction with normal on-ramps here? I would have thought that many on-ramps / off ramps exist that are not necessarily slight turns.
Does the output produce a lot of information here? I would have restricted the check for turns between ramps (which is not possible to check based on the turn type).

What do you think, should we give a check comparing ramp-classes for sharp turns a try?